src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer.Common/BackgroundDocumentGenerator.cs

```diff
@@ -172,7 +170,7 @@ private async void Timer_Tick(object state)
                     var document = work[i].Value;
                     try
                     {
-                        await document.GetGeneratedOutputAsync();
```

Comment:
Ahh these two aren't the same. You still need to await this to ensure code following it doesn't start running. The pattern you can use here is at the containing method level. Aka:

```C#
private void Timer_Tick(object state)
{
    _ = Timer_TickAsync(CancellationToken.None);
}

private async Task Timer_TickAsync(CancellationToken cancellationToken)
{
// ... The old `Timer_Tick` content with the awaits
}
```

Same in other places in this PR

Reply:
I'm not super clear on what happened here but it seems like GitHub is not displaying what is actually reflected on disk? I'm gonna rebase and see if that helps.

Reply:
Oh, there are two files with the same name but in different folders, nevermind, will fix.